addons/mrp/models/mrp_production.py

```diff
@@ -1454,77 +1452,221 @@ def _get_name_backorder(self, name, sequence):
-        if not close_mo:
-            self.move_raw_ids.filtered(lambda m: not m.additional)._do_unreserve()
-            self.move_raw_ids.filtered(lambda m: not m.additional)._action_assign()
-        backorders.action_confirm()
-        backorders.action_assign()
-
-        # Remove the serial move line without reserved quantity. Post inventory will assigned all the non done moves
-        # So those move lines are duplicated.
-        backorders.move_raw_ids.move_line_ids.filtered(lambda ml: ml.product_id.tracking == 'serial' and ml.reserved_qty == 0).unlink()
-
-        for old_wo, wo in zip(self.workorder_ids, backorders.workorder_ids):
-            wo.qty_produced = max(old_wo.qty_produced - old_wo.qty_producing, 0)
-            if wo.product_tracking == 'serial':
-                wo.qty_producing = 1
-            else:
-                wo.qty_producing = wo.qty_remaining
+            production.name = self._get_name_backorder(
+                production.name, production.backorder_sequence)
+            production.product_qty = count[production][0]
+            backorder_vals = production.copy_data(
+                default=production._get_backorder_mo_vals())[0]
+            backorder_qty_list = count[production][1:]
+
+            next_seq = max(production.procurement_group_id.mrp_production_ids.mapped(
+                "backorder_sequence"), default=1)
+
+            for qty_to_backorder in backorder_qty_list:
+                next_seq += 1
+                backorder_vals_list.append(dict(
+                    backorder_vals,
+                    product_qty=qty_to_backorder,
+                    name=production._get_name_backorder(
+                        production.name, next_seq),
+                    backorder_sequence=next_seq
+                ))
+
+        backorders = self.env['mrp.production'].create(backorder_vals_list)
 
-        return backorders
+        # TODO maybe remove that in bellow loop. Code readibilty <-> perf
+        index = 0
+        production_to_backorders = {}
+        production_ids = OrderedSet()
+        for production in self:
+            number_of_backorder_created = len(
+                count.get(production, _default_count(production))) - 1
+            backorders_from_production = backorders[index:index + number_of_backorder_created]
+            production_to_backorders[production] = backorders_from_production
+            production_ids.update(production.ids)
+            production_ids.update(backorders_from_production.ids)
+            index += number_of_backorder_created
+
+        # Split the `stock.move` among new backorders.
+        new_moves_vals = []
+        moves = []
+        for production in self:
+            for move in production.move_raw_ids | production.move_finished_ids:
+                if move.additional:
+                    continue
+                unit_factor = move.product_qty / initial_qty_by_production[production]
```

Comment:
Instead of `move.product_qty`, can we use `move.product_uom_qty` to avoid the uom conversion in `_get_move_product_uom_qty`? If yes, then we don't need `_get_move_product_uom_qty` do we?